### DISCUSSION TITLE
I think `ensureSuspenseTimers` should ALWAYS set staleTime to 1000 when it is below 1000.

### DISCUSSION THREAD

**Discussion by stakbucks** [TanStack/query#8366] [General] [answered]
I created my QueryClient like this:
```typescript
new QueryClient({
    defaultOptions: {
      queries: {
        staleTime: 10,
        retry: 0,
      },
      dehydrate: {
        shouldDehydrateQuery: (query) =>
          defaultShouldDehydrateQuery(query) ||
          query.state.status === "pending",
      },
    },
  });
```

I was using `Suspense`+ `useSuspenseQuery` in my app, and I knew that when using `useSuspenseQuery`, it would always prevent fetching again when directly mounting after suspending. However, I found out that most of the time, it would fetch again after suspending.
I was really confused and thought that this was a bug, so I looked up the code for `useSuspenseQuery` to see what was happening and found out that it ONLY set `staleTime` to 1000 when `staleTime` value was `undefined`. I think this should work unintended for some edge cases like mine, when someone sets really small `staleTime`, and the rendering took longer than the `staleTime`.
So, I think [ensureSuspenseTimers](https://github.com/TanStack/query/blob/c402a2b815c97292bd76b7516242c30806c7efef/packages/react-query/src/suspense.ts#L21), should always set `staleTime` to 1000, when its value is below 1000. What do you think?

**Comment by TkDodo** [comment, marked as answer]
yep; PRs welcome
